source/code/plugins/out_oms_changetracking_file.rb

```diff
@@ -0,0 +1,591 @@
+        @@PrimaryContentLocation = primaryContentLocation
+    end 
+    def get_SecondryContentLocation
+       return @@SecondryContentLocation
+    end 
+    def set_SecondryContentLocation(secondryContentLocation)
+        @@SecondryContentLocation = secondryContentLocation
+    end 
+
+    def configure(conf)
+      s = conf.add_element("secondary")
+      s["type"] = ChunkErrorHandler::SecondaryName
+
+      super
+      if !@PrimaryContentLocation.nil? and !@PrimaryContentLocation.empty? and @PrimaryContentLocation.include? "http" or @PrimaryContentLocation.include? "https"
+         urlDetails = @PrimaryContentLocation.split('?')
+         if !urlDetails.nil? and urlDetails.length == 2
+            @@ContentlocationUri = urlDetails[0].strip
+            @@PrimaryContentLocationAccessToken = urlDetails[1]
+         end
+      end
+      if !@SecondryContentLocation.nil? and !@SecondryContentLocation.empty? and @SecondryContentLocation.include? "http" or @SecondryContentLocation.include? "https"
+         urlDetails = @SecondryContentLocation.split('?')
+         if !urlDetails.nil? and urlDetails.length == 2
+            @@SecondryContentLocationAccessToken = urlDetails[1].strip
+         end
+      end
+    end
+
+    def start
+      super
+      @proxy_config = OMS::Configuration.get_proxy_config(@proxy_conf_path)
+      @sha256 = Digest::SHA256.new
+      @log.debug "buffer_path : #{@buffer_path}"
+      if !@buffer_path.empty?
+         contentlocationfilepath = File.dirname(@buffer_path) + '/' + @@ContentLocationCacheFileName
+
+         @log.debug "contentlocationfilepath : #{contentlocationfilepath}"
+         if File.exists?(contentlocationfilepath)
+            @@LastContentlocationUri = File.open(contentlocationfilepath, &:gets).strip
+         end
+      end
+      @log.debug "LastContentlocationUri : #{@@LastContentlocationUri}"
+    end
+
+    def shutdown
+      if !@buffer_path.empty?
+         contentlocationfilepath = File.dirname(@buffer_path) + '/' + @@ContentLocationCacheFileName
+         if File.exists?(contentlocationfilepath)
+            File.open(contentlocationfilepath, "w") do |f|
+                f.puts "#{@@ContentlocationUri}"
+            end 
+         else
+            File.write(contentlocationfilepath, "#{@@ContentlocationUri}")
+         end
+      end
+      @log.debug "LastContentlocationUri written to : #{contentlocationfilepath}"
+      super
+    end
+
+    ####################################################################################################
+    # Methods
+    ####################################################################################################
+
+    # create a HTTP request to PUT blob
+    # parameters:
+    #   uri: URI. blob URI
+    #   msg: string. body of the request
+    #   file_path: string. file path
+    # returns:
+    #   HTTPRequest. blob PUT request
+    def create_blob_put_request(uri, msg, request_id, file_path = nil)
```

Comment:
This feature required complex functionality in the out plugin.
Basically, this output plugin needs to these functions

Create two blob url based on config parameters
extract changed files from records and Upload files to blob
Upload to blob should happen without gets (security reasons)
Do failsafe if one url fails.
Send the record to ODS
Send a notification to ODS on failures.
there is no match rule that can do this. we have scattered functionality in out_oms_blob and out_oms, but I cant call methods in those, due to (3).

its better to create a new match that works only for file change tracking and can be maintained independent of others